pydoctor/test/test_templatewriter.py

```diff
@@ -484,6 +484,31 @@ class _Private(_BaseForPrivate):
     assert not isClassNodePrivate(cast(model.Class, mod.contents['_BaseForPublic']))
     assert isClassNodePrivate(cast(model.Class, mod.contents['_BaseForPrivate']))
 
+@systemcls_param
+def test_format_function_def_overloads(systemcls: Type[model.System]) -> None:
+    mod = fromText("""
+        from typing import overload, Union
+        @overload
+        def parse(s: str) -> str:
+            ...
+        @overload
+        def parse(s: bytes) -> bytes:
+            ...
+        def parse(s: Union[str, bytes]) -> Union[str, bytes]:
+            pass
+        """, systemcls=systemcls)
+    func = mod.contents['parse']
+    assert isinstance(func, model.Function)
+    
+    # We intentionally remove spaces before comparing
+    overloads_html = stanutils.flatten(list(pages.format_overloads(func))).replace(' ','')
+    assert '''(s:str)-&gt;str:''' in overloads_html
+    assert '''(s:bytes)-&gt;bytes:''' in overloads_html
+
+    # Confirm the actual function definition is not rendered
+    function_def_html = stanutils.flatten(list(pages.format_function_def(func.name, func.is_async, func)))
```

Comment:
Isn't it a bit weird that the format_function_def() returns nothing for overloads ? I believe it's OK, but should be documented in the docstring.